### PR TITLE
Change runtime measurement method

### PR DESCRIPTION
**Pull request by liamplm**
https://github.com/soallpeach/onboarding/issues/23

I think using the docker container state for measuring execution time is a better option compare to the current method that measures container creation and destruction time
So I used `State.StartedAt` and `State.FinishedAt` that are available by `docker inspect <container-name>`

thanks for this cool contest !!

### CODE REVIEW

**Comment by s-soroosh** [comment]
Hi @liamplm, 
Thank you so much for your contribution to this project.  Would you mind to open this PR against branch `stdout-elapsed-time`? 
Then I can test it locally to ensure all works as expected.

File: run.py [hunk 1]
```diff
@@ -41,6 +40,14 @@ def run_step(self, name: str, script_path: str, timeout: int = 10 * 60) -> StepR
         print(stderr)
         print(stdout)
 
+        if not get_durtion_from_stdout:
+            elapsed = time.time()
+            duration = round(elapsed - start, 3)
+        else:
+            try:
+                duration = float(stdout)
```
Comment (s-soroosh) [inline run.py]: The implementation is risky as there might be some other stuff on stdout as well. 
Let's do it this way:
Expect there would a line in the stdout with this format: 
`::::DURATION=value` 
find the latest occurrence of it and extract the `value` out of it.

File: run.py [hunk 2]
```diff
@@ -41,6 +40,14 @@ def run_step(self, name: str, script_path: str, timeout: int = 10 * 60) -> StepR
...
+            except ValueError:
+                duration = None
```
Comment (s-soroosh) [inline run.py:50]: If there was a value error:
print a warning and then fall back to the normal duration calculation model.

File: run.py (hunk 1 above)
Comment (liamplm) [reply run.py]: Yes, I totally agree ... but what about a JSON output?

File: scripts/run_in_file_program.sh [hunk 3]
```diff
@@ -3,8 +3,22 @@ BASE_PATH=$(pwd)
...
+
+START_DATE=$(docker inspect --format='{{.State.StartedAt}}' "$CHALLENGE_NAME-container")
+STOP_DATE=$(docker inspect --format='{{.State.FinishedAt}}' "$CHALLENGE_NAME-container")
+
+START_TIMESTAMP=$(date --date=$START_DATE +'%s.%3N')
+STOP_TIMESTAMP=$(date --date=$STOP_DATE +'%s.%3N')
+
+docker container rm "$CHALLENGE_NAME-container" &> /dev/null
 
+echo $(bc <<< "$STOP_TIMESTAMP - $START_TIMESTAMP")
```
Comment (s-soroosh) [inline scripts/run_in_file_program.sh]:
```suggestion
echo "::::DURATION="$(bc <<< "$STOP_TIMESTAMP - $START_TIMESTAMP")
```

File: run.py (hunk 1 above)
Comment (s-soroosh) [reply run.py]: I can't think of any situation we would need more than one key-value per line. 
+ generating JSON in the shell is quite counterintuitive.

**Comment by liamplm** [comment]
@psycho-ir Isn't it better if we use the average of like 5 times of running a container?

**Comment by s-soroosh** [comment]
> @psycho-ir Isn't it better if we use the average of like 5 times of running a container?

We can, later on, show the average of runs in day to the user.  but for now, I think what we have is good enough.

**Comment by liamplm** [comment]
> > @psycho-ir Isn't it better if we use the average of like 5 times of running a container?
> 
> We can, later on, show the average of runs in day to the user. but for now, I think what we have is good enough.


I meant using an average function every time we run a program ... I already implement that and if it`s ok to you I will put it in my next pull request

**Comment by s-soroosh** [comment]
@liamplm  Yes I got your point, I don't feel like we would need. especially since the next challenges probably would have different evaluation model. 
Let's see after 2nd challenge what we can improve.

**Pull request merged**